ere-guests/libs/src/block_ssz.rs

```diff
@@ -0,0 +1,568 @@
+            ommers: body.ommers.into_iter().map(Header::from).collect(),
+            withdrawals: body
+                .withdrawals
+                .map(|ws| ws.into_iter().map(Withdrawal::from).collect()),
+        }
+    }
+}
+
+/// SSZ-serializable representation of an Ethereum block header.
+#[derive(Debug, PartialEq, Eq, ssz_derive::Encode, ssz_derive::Decode)]
+pub struct Header {
+    parent_hash: B256,
+    ommers_hash: B256,
+    beneficiary: Address,
+    state_root: B256,
+    transactions_root: B256,
+    receipts_root: B256,
+    logs_bloom: Bloom,
+    difficulty: U256,
+    number: BlockNumber,
+    gas_limit: u64,
+    gas_used: u64,
+    timestamp: u64,
+    extra_data: Bytes,
+    mix_hash: B256,
+    nonce: B64,
+    base_fee_per_gas: Option<u64>,
+    withdrawals_root: Option<B256>,
+    blob_gas_used: Option<u64>,
+    excess_blob_gas: Option<u64>,
+    parent_beacon_block_root: Option<B256>,
+    requests_hash: Option<B256>,
+}
+
+impl From<alloy_consensus::Header> for Header {
+    fn from(header: alloy_consensus::Header) -> Self {
+        Self {
+            parent_hash: header.parent_hash,
+            ommers_hash: header.ommers_hash,
+            beneficiary: header.beneficiary,
+            state_root: header.state_root,
+            transactions_root: header.transactions_root,
+            receipts_root: header.receipts_root,
+            logs_bloom: header.logs_bloom,
+            difficulty: header.difficulty,
+            number: header.number,
+            gas_limit: header.gas_limit,
+            gas_used: header.gas_used,
+            timestamp: header.timestamp,
+            extra_data: header.extra_data,
+            mix_hash: header.mix_hash,
+            nonce: header.nonce,
+            base_fee_per_gas: header.base_fee_per_gas,
+            withdrawals_root: header.withdrawals_root,
+            blob_gas_used: header.blob_gas_used,
+            excess_blob_gas: header.excess_blob_gas,
+            parent_beacon_block_root: header.parent_beacon_block_root,
+            requests_hash: header.requests_hash,
+        }
+    }
+}
+
+/// SSZ-serializable representation of a validator withdrawal.
+#[derive(Debug, PartialEq, Eq, ssz_derive::Encode, ssz_derive::Decode)]
+pub struct Withdrawal {
+    index: u64,
+    validator_index: u64,
+    address: Address,
+    amount: u64,
+}
+
+impl From<eip4895::Withdrawal> for Withdrawal {
+    fn from(withdrawal: eip4895::Withdrawal) -> Self {
+        Self {
+            index: withdrawal.index,
+            validator_index: withdrawal.validator_index,
+            address: withdrawal.address,
+            amount: withdrawal.amount,
+        }
+    }
+}
+
+/// SSZ-serializable transaction envelope supporting a subset of Ethereum transaction types.
+#[derive(Debug, PartialEq, Eq, ssz_derive::Encode, ssz_derive::Decode)]
+#[ssz(enum_behaviour = "union")]
+pub enum EthereumTxEnvelope {
+    /// Legacy transaction type.
+    Legacy(SignedTx<TxLegacy>),
```

Comment:
First I wanted to only support legacy transaction since is the only type used in EEST -- so I didn't had to bloat this module with everything.

But later I realized that for realistic  benchmark comparisons we would like to run with mainnet blocks -- doing an early run realized I needed all, so had to go "all in". NBD, but just a bit more bloat.